src/cmdhandler/command_handler.cpp

```diff
@@ -191,18 +197,38 @@ std::string command_handler::state_system(const std::vector<std::string>& tokens
         std::string state_to_delete = tokens[2];
 
         //if value is 'current', clear out current state
-        //if not, attempt to delete save state's file
+        //if not, attempt to delete given save state's file
         if(state_to_delete == "current"){
             current_state.robots.clear();
             current_state.collectors.clear();
             return "current state has been cleared";
         }else{
-            if(remove(state_to_delete.c_str()) != 0){
+            if(remove(("states/"+state_to_delete).c_str()) != 0){
```

Comment:
This is up to you if you want to do it or not, but you can use C++17's `filesystem` standard library for deleting a file: https://en.cppreference.com/w/cpp/filesystem/remove